src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -182,6 +182,12 @@ internal async Task IndexMethodAsyncCore(MethodInfo method, IFunctionIndexCollec
                     {
                         throw new InvalidOperationException("More than one trigger per function is not allowed.");
                     }
+
+                    if ((method.GetCustomAttribute<RetryAttribute>() != null || method.DeclaringType.GetCustomAttribute<RetryAttribute>() != null) &&
+                        triggerBinding.GetType().GetCustomAttribute<SupportsRetryAttribute>() == null)
+                    {
+                        _logger?.LogWarning($"Retries are not allowed for {triggerBinding.GetType().Name}");
```

Comment:
nit: the trigger binding type is internal, so shouldn't be logged to the user. Instead just say something like "Retries are not supported by the trigger binding for <function>."